# **RETAIL STORE SALES**

## **Import Library**

In [2]:
import pandas as pd

## **Memuat Data**

In [7]:
file = "/content/drive/MyDrive/PRAKTIKUM AVD /retail_store_sales.csv"
df = pd.read_csv(file)

df

Transaction ID Customer ID       Category          Item  Price Per Unit  \
0        TXN_6867343     CUST_09     Patisserie   Item_10_PAT            18.5   
1        TXN_3731986     CUST_22  Milk Products  Item_17_MILK            29.0   
2        TXN_9303719     CUST_02       Butchers   Item_12_BUT            21.5   
3        TXN_9458126     CUST_06      Beverages   Item_16_BEV            27.5   
4        TXN_4575373     CUST_05           Food   Item_6_FOOD            12.5   
...              ...         ...            ...           ...             ...   
12570    TXN_9347481     CUST_18     Patisserie   Item_23_PAT            38.0   
12571    TXN_4009414     CUST_03      Beverages    Item_2_BEV             6.5   
12572    TXN_5306010     CUST_11       Butchers    Item_7_BUT            14.0   
12573    TXN_5167298     CUST_04      Furniture    Item_7_FUR            14.0   
12574    TXN_2407494     CUST_23           Food   Item_9_FOOD            17.0   

       Quantity  Total Spent  Payment Method  Location Transaction Date  \
0          10.0        185.0  Digital Wallet    Online       2024-04-08   
1           9.0        261.0  Digital Wallet    Online       2023-07-23   
2           2.0         43.0     Credit Card    Online       2022-10-05   
3           9.0        247.5     Credit Card    Online       2022-05-07   
4           7.0         87.5  Digital Wallet    Online       2022-10-02   
...         ...          ...             ...       ...              ...   
12570       4.0        152.0     Credit Card  In-store       2023-09-03   
12571       9.0         58.5            Cash    Online       2022-08-12   
12572      10.0        140.0            Cash    Online       2024-08-24   
12573       6.0         84.0            Cash    Online       2023-12-30   
12574       3.0         51.0            Cash    Online       2022-08-06   

      Discount Applied  
0                 True  
1                 True  
2                False  
3                  NaN  
4                False  
...                ...  
12570              NaN  
12571            False  
12572              NaN  
12573             True  
12574              NaN  

[12575 rows x 11 columns]

In [8]:
df.head(10)

Transaction ID Customer ID       Category          Item  Price Per Unit  \
0    TXN_6867343     CUST_09     Patisserie   Item_10_PAT            18.5   
1    TXN_3731986     CUST_22  Milk Products  Item_17_MILK            29.0   
2    TXN_9303719     CUST_02       Butchers   Item_12_BUT            21.5   
3    TXN_9458126     CUST_06      Beverages   Item_16_BEV            27.5   
4    TXN_4575373     CUST_05           Food   Item_6_FOOD            12.5   
5    TXN_7482416     CUST_09     Patisserie           NaN             NaN   
6    TXN_3652209     CUST_07           Food   Item_1_FOOD             5.0   
7    TXN_1372952     CUST_21      Furniture           NaN            33.5   
8    TXN_9728486     CUST_23      Furniture   Item_16_FUR            27.5   
9    TXN_2722661     CUST_25       Butchers   Item_22_BUT            36.5   

   Quantity  Total Spent  Payment Method  Location Transaction Date  \
0      10.0        185.0  Digital Wallet    Online       2024-04-08   
1       9.0        261.0  Digital Wallet    Online       2023-07-23   
2       2.0         43.0     Credit Card    Online       2022-10-05   
3       9.0        247.5     Credit Card    Online       2022-05-07   
4       7.0         87.5  Digital Wallet    Online       2022-10-02   
5      10.0        200.0     Credit Card    Online       2023-11-30   
6       8.0         40.0     Credit Card  In-store       2023-06-10   
7       NaN          NaN  Digital Wallet  In-store       2024-04-02   
8       1.0         27.5     Credit Card  In-store       2023-04-26   
9       3.0        109.5            Cash    Online       2024-03-14   

  Discount Applied  
0             True  
1             True  
2            False  
3              NaN  
4            False  
5              NaN  
6             True  
7             True  
8            False  
9            False

In [9]:
df.tail(10)

Transaction ID Customer ID                            Category  \
12565    TXN_1712895     CUST_07  Computers and electric accessories   
12566    TXN_6298574     CUST_01                       Milk Products   
12567    TXN_4890318     CUST_25                            Butchers   
12568    TXN_1096134     CUST_06                                Food   
12569    TXN_2318174     CUST_18                          Patisserie   
12570    TXN_9347481     CUST_18                          Patisserie   
12571    TXN_4009414     CUST_03                           Beverages   
12572    TXN_5306010     CUST_11                            Butchers   
12573    TXN_5167298     CUST_04                           Furniture   
12574    TXN_2407494     CUST_23                                Food   

              Item  Price Per Unit  Quantity  Total Spent  Payment Method  \
12565   Item_2_CEA             6.5       5.0         32.5  Digital Wallet   
12566  Item_1_MILK             5.0       4.0         20.0            Cash   
12567   Item_3_BUT             8.0       5.0         40.0     Credit Card   
12568  Item_4_FOOD             9.5       8.0         76.0  Digital Wallet   
12569   Item_1_PAT             5.0       9.0         45.0  Digital Wallet   
12570  Item_23_PAT            38.0       4.0        152.0     Credit Card   
12571   Item_2_BEV             6.5       9.0         58.5            Cash   
12572   Item_7_BUT            14.0      10.0        140.0            Cash   
12573   Item_7_FUR            14.0       6.0         84.0            Cash   
12574  Item_9_FOOD            17.0       3.0         51.0            Cash   

       Location Transaction Date Discount Applied  
12565  In-store       2024-09-06            False  
12566    Online       2023-05-09            False  
12567  In-store       2023-03-28            False  
12568  In-store       2022-06-11              NaN  
12569  In-store       2024-05-13            False  
12570  In-store       2023-09-03              NaN  
12571    Online       2022-08-12            False  
12572    Online       2024-08-24              NaN  
12573    Online       2023-12-30             True  
12574    Online       2022-08-06              NaN

# **Business Understanding**

## **Business Objective**

Dataset berisi data transaksi penjualan ritel, mencakup informasi seperti:

* ID transaksi & pelanggan

* Kategori dan nama item

* Harga per unit & jumlah pembelian

* Total belanja

* Metode pembayaran

* Lokasi (online/offline)

* Tanggal transaksi

* Status diskon

Dataset ini menggambarkan perilaku belanja pelanggan dalam sebuah toko ritel, baik dari sisi produk, nilai transaksi, hingga cara pembayaran.

Dataset ini memiliki 12575 baris dan 11 kolom.

**Studi Kasus**

Sebuah perusahaan ritel ingin meningkatkan pendapatan dan memahami pola belanja pelanggan. Mereka mengumpulkan data transaksi untuk dianalisis agar tahu:

* Produk apa paling laku

* Kapan penjualan tinggi atau rendah

* Apakah diskon efektif

* Metode pembayaran favorit pelanggan



## **Assess Situation**

**Kondisi Dataset**

* Dataset berisi 12.575 baris data sintetis yang merepresentasikan transaksi penjualan di toko ritel

* Terdapat 8 kategori produk, masing-masing memiliki 25 item dengan harga statis

* Data dirancang untuk mensimulasikan kondisi penjualan dunia nyata, sehingga cukup realistis untuk dianalisis

* Variabel beragam numerik dan kategori untuk mendukung analisis

* Terdapat kolom waktu transaksi yang memungkinkan analisis tren penjualan

* Dataset memang ditujukan untuk latihan data cleaning, EDA, dan feature engineering

**Tantangan Dataset**

* Dataset sengaja dibuat kotor

* Dari dataset ini perlu dibersihkan terlebih dahulu


## **Analytic Goals & Project Plan**

**Tujuan**

Mengolah dataset ini adalah menganalisis data penjualan ritel untuk memahami pola penjualan dan  preferensi pelanggan terhadap produk dan metode pembayaran.

**Rencana**

* Memahami data dengan mengetahui tipe data dan memahami struktur kategori serta item produk

* Mengolah data dengan menangani data yang kosong dan memperbaiki data tidak konsisten.

* melakukan EDA dengan mencari kategori terlaris, melihat distribusi belanja, menganalisis metode pembayaran, dan mengamati tren penjualan.

* Membuat visualisasi data.

* Menarik kesimpulan dan rekomendasi untuk menentukan strategi promosi dan meningkatkan penjualan.